index.js

```diff
@@ -43,17 +43,37 @@ function logGraphQLDetails (opts, schema, document, context) {
   const queryOps = readOps(document, 'query', opts)
   const mutationOps = readOps(document, 'mutation', opts)
 
+  // operationName can be provided at the request level, or it can be provided by query/mutation.
+  // If it's provided by both, the request level takes precedence.
+  const operationName = context.reply.request.body.operationName || readOperationName(document, opts)
+  const isCurrentOperation = (op) => op.operationName === operationName
+
+  // Runs on a single operation at a time in a batched query, so we need to pull out
+  // the relevant operation from the batch to be able to log variables for it.
+  const isBatch = Array.isArray(context.reply.request.body)
+  const currentBody = isBatch
```

Comment:
we can skip the find operation when we don't want to log anything

```suggestion
  const detailedLog = opts.logVariables || opts.logBody
  const currentBody = detailedLog && isBatch
```